lib/dolly/mango.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Dolly
+  module Mango
+
+    SELECTOR_SYMBOL = "$"
+    AVAILABLE_SELECTORS = %I[
+      lt
+      lte
+      eq
+      ne
+      gte
+      gt
+      exists
+      type
+      in
+      nin
+      size
+      mod
+      regex
+    ].freeze
+
+    DESIGN = "_find"
+
+    def find_by(query, opts = {})
+      opts.merge!(limit: 1)
+      perform_query(build_query(query, opts))[:docs].first
+    end
+
+    def where(query, opts = {})
+      perform_query(build_query(query, opts))[:docs]
```

Comment:
same as with find, should we return initialized models? or at least give the option to do so?

Reply:
`perform_query(build_query(query, opts))` what will this return in an empty result search? Will it be a hash with `:docs` empty? or ???

Reply:
this returns an empty array